### PR TITLE
Spring boot branch

### PR DESCRIPTION
**Pull request by KajalMathur**
Please review address comments.

### CODE REVIEW

File: src/main/java/com/product/service/ProductServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import com.product.dao.ProductDaoImpl;
+import com.product.model.Product;
+
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+@Data
```
Comment (AakashIsTheLimit) [inline src/main/java/com/product/service/ProductServiceImpl.java:16]: This is Service class, you dont need it to have `@Data` (Service class should never have @data)

File: src/main/java/com/product/service/ProductServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+@Slf4j
+@Service
+public class ProductServiceImpl implements ProductService {
+
+	@Autowired
+	private ProductDaoImpl productDaoImpl;
```
Comment (AakashIsTheLimit) [inline src/main/java/com/product/service/ProductServiceImpl.java:22]: No field level injection. Always constructor injection.

File: src/main/java/com/product/service/ProductServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+	@Value("${spring.application.id}")
+	private String propId;
+
+	public ProductServiceImpl(ProductDaoImpl productDaoImpl) {
```
Comment (AakashIsTheLimit) [inline src/main/java/com/product/service/ProductServiceImpl.java:27]: For it to be construction injection, there should be @Autowired on top of constructor, not the field.

File: src/main/java/com/product/service/ProductService.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package com.product.service;
+
+import java.util.Collection;
+import java.util.Optional;
+
+import com.product.model.Product;
+
+public interface ProductService {
+
+	public Collection<Product> getAllProducts();
```
Comment (AakashIsTheLimit) [inline src/main/java/com/product/service/ProductService.java:10]: Why Collection (superclass of List).. It should be List instead

Read into Collection, List, ArrayList and whats the difference between all

File: src/main/java/com/product/dao/ProductDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+import com.product.model.Customer;
+import com.product.model.Product;
+
+@Repository
+public class ProductDaoImpl implements ProductDao {
+
+	private DefaultProductDao productDaoDefault;
+
+	@Autowired
```
Comment (AakashIsTheLimit) [inline src/main/java/com/product/dao/ProductDaoImpl.java:18]: constructor autowire this

File: src/main/java/com/product/dao/ProductDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+	RestTemplate restTemplate;
+
+	public ProductDaoImpl(DefaultProductDao productDaoDefault) {
```
Comment (AakashIsTheLimit) [inline src/main/java/com/product/dao/ProductDaoImpl.java:21]: autowire on top of this.

File: src/main/java/com/product/dao/DefaultProductDao.java [hunk 7]
```diff
@@ -0,0 +1,9 @@
+package com.product.dao;
```
Comment (AakashIsTheLimit) [inline src/main/java/com/product/dao/DefaultProductDao.java:1]: this file is actually not needed. I know what might have inspired this. Lets chat offline.

There should just be 1 interface (ProductDao) class not two (DefaultProductDao, ProductDao)

and the implementation class `ProductDaoImpl` should implement `ProductDao` just like you did in CustomerService. 

And for getting the customer, that shoudl be part of another CustomerDao and CustomerDaoImpl, it does not belong to `ProductDao` .. ProductDao should only deal with getting, adding, removing, updating Products.

CustomerDao should be dealing with just Customer.

**Review by AakashIsTheLimit — CHANGES_REQUESTED** [review]

File: src/main/java/com/product/service/ProductServiceImpl.java (hunk 1 above)
Comment (KajalMathur) [reply src/main/java/com/product/service/ProductServiceImpl.java:16]: Done.

File: src/main/java/com/product/service/ProductServiceImpl.java (hunk 2 above)
Comment (KajalMathur) [reply src/main/java/com/product/service/ProductServiceImpl.java:22]: Done.

File: src/main/java/com/product/service/ProductServiceImpl.java (hunk 3 above)
Comment (KajalMathur) [reply src/main/java/com/product/service/ProductServiceImpl.java:27]: Removed.

File: src/main/java/com/product/service/ProductService.java (hunk 4 above)
Comment (KajalMathur) [reply src/main/java/com/product/service/ProductService.java:10]: Done.

File: src/main/java/com/product/dao/ProductDaoImpl.java (hunk 5 above)
Comment (KajalMathur) [reply src/main/java/com/product/dao/ProductDaoImpl.java:18]: Done.

File: src/main/java/com/product/dao/ProductDaoImpl.java (hunk 6 above)
Comment (KajalMathur) [reply src/main/java/com/product/dao/ProductDaoImpl.java:21]: Done.

File: src/main/java/com/product/dao/DefaultProductDao.java (hunk 7 above)
Comment (KajalMathur) [reply src/main/java/com/product/dao/DefaultProductDao.java:1]: As MongoRepository is Interface that's why i opted this option.

**Comment by KajalMathur** [comment]
All the changes has done.

**Pull request closed without merging**